### DISCUSSION TITLE
Where can I find deploymentId and futureId when wiping a previous execution?

### DISCUSSION THREAD

**Discussion by zhibo-noumena** [NomicFoundation/hardhat#5005] [Troubleshooting]
Hello,

I am following https://hardhat.org/ignition/docs/guides/error-handling#wiping-a-previous-execution to wipe an execution? However, I don't see where to find the **deploymentId** and **futureId**.

Can you suggest which folders to find these two parameters?

Thank you!

**Comment by kanej** [comment]
Hello @zhibo-noumena.
The `deployment-id` identifies a particular deployment. Whenever you run `npx hardhat ignition deploy` it is done as part of a deployment. You can pass a `deployment-id` with the `--deployment-id` parameter. If none is passed, a default `deployment-id` is used matching the template `chain-<chain-id>` so `chain-31337` for the local Hardhat node.

The future id is the id of the future you are attempting to wipe. A future will be something like a contract deploy. Typically you are wiping a future because it errored or you have modified it, say updating a contracts code (and therefore bytecode) to add new constructor arguments. Ignition should show an error asking you to wipe the futures state as a way of being explicit that you intend to rerun the entire contract deployment.

What where you trying to achieve in wiping the previous execution. Was the goal to wipe an entire deployment?

Deployments are stored under `./ignition/deployments/<deployment-id>`.
